host-spec/c04-networking.tm

```diff
@@ -1,95 +1,244 @@
-  to establish and maintain a list of peers and to share that list with other
-  peers from the network.
+  Each Polkadot Host node maintains a ED25519 key pair which is used to
+  identify the node. The public key is shared with the rest of the network
+  which allows nodes to establish secure communication channels.\ 
 
-  The Polkadot Host uses various mechanism for peer dicovery.
+  Each node must have its own unique ED25519 key pair. When two or more nodes
+  use the same key, the network will interpret those nodes as a single node,
+  which will result in undefined behavior and can result in equivocation.
+  Furthermore, the node's <verbatim|PeerId> as defined in Definition
+  <reference|defn-peer-id> is derived from its public key. <verbatim|PeerId>
+  (<reference|defn-peer-id>) is used to identify each node when they are
+  discoved in the course of the discovery mechanism described in Section
+  <reference|sect-discovery-mechanism>.
 
-  <\itemize>
-    <item>Bootstrap nodes - hard-coded node identities and addresses provided
-    by network configuration itself. Those addresses are selected an updated
-    by the developers of the Polkadot Host. Node addresses should be selected
-    based on a reputation metric, such as reliability and uptime.
+  <\definition>
+    <label|defn-peer-id>The Polkadot node's <verbatim|PeerId>, formally
+    referred to as <math|P<rsub|id>>, is derived from the ED25519 public key
+    and is structured as defined in the libp2p specification
+    (<slink|https://docs.libp2p.io/concepts/peer-id/>).
+  </definition>
+
+  <subsection|Discovery mechanism><label|sect-discovery-mechanism>
 
-    <item>mDNS - performs a broadcast to the local network. Nodes that might
-    be listing can respond the the broadcast.
+  The Polkadot Host uses various mechanisms to find peers within the network,
+  to establish and maintain a list of peers and to share that list with other
+  peers from the network, as follows:
 
-    <item>Kademlia requests - Kademlia supports <verbatim|FIND_NODE>
+  <\itemize>
+    <item><strong|Bootstrap nodes> are hard-coded node identities and
+    addresses provided by the genesis state specification as described in
+    Appendix <reference|sect-genesis-block>.
+
+    <item><strong|mDNS> protocol which performs a broadcast to the local
+    network. Nodes that might be listening can respond the the broadcast.
+    <hlink|The libp2p mDNS specification|https://github.com/libp2p/specs/blob/master/discovery/mdns.md>
+    defines this process in more detail. This protocol is an optional
+    implementation detail for Polkadot Host implementers and is not required
+    in order to participate in the Polkadot network.
+
+    <item><strong|Kademlia requests> invoking Kademlia <verbatim|FIND_NODE>
     requests, where nodes respond with their list of available peers.
+    Kademlia requests must also contain protocol identifiers as described in
+    Section <reference|sect-protocol-identifier>.
```

Comment:
This reference does not exists. Do you maybe mean `sect-protocols-substreams`?

Reply:
Fixed:

![image](https://user-images.githubusercontent.com/42901763/106746633-14930500-6623-11eb-959f-50dcfa893108.png)